### DISCUSSION TITLE
Sometimes we should avoid using bitpacked encoding

### DISCUSSION THREAD

**Discussion by doki23** [vortex-data/vortex#1654] [Feature Requests]
For instance, the bit_width is 7 and the type is u8, the benefit is so small that we can return an Err to avoid encoding.

**Comment by gatesn** [comment]
Yup, completely agree. We need to introduce some concept of this into the objective function of the compressor. A sufficiently small ratio should be penalised for slowing down decompression throughput.

**Comment by robert3005** [comment]
This now happens, we don't unconditionally bitpack unless the target size is smaller
